src/widgets/allergies/allergies-detailed-summary.component.tsx

```diff
@@ -21,100 +25,91 @@ export default function AllergiesDetailedSummary(
   ] = useCurrentPatient();
   const match = useRouteMatch();
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (!isLoadingPatient && patient) {
-      const abortController = new AbortController();
+      const sub = performPatientAllergySearch(
+        patient.identifier[0].value
+      ).subscribe(allergies => {
+        setPatientAllergies(allergies), createErrorHandler();
+      });
 
-      performPatientAllergySearch(patient.identifier[0].value, abortController)
-        .then(allergy => setPatientAllergy(allergy.data))
-        .catch(createErrorHandler());
-
-      return () => abortController.abort();
+      return () => sub.unsubscribe();
     }
   }, [isLoadingPatient, patient]);
 
-  function displayAllergy() {
-    return (
-      <SummaryCard
-        name="Allergies"
-        styles={{ width: "100%" }}
-        addComponent={AllergyForm}
-        showComponent={() =>
-          openWorkspaceTab(AllergyForm, "Allergy Form", {
-            allergyUuid: null
-          })
-        }
-      >
-        <table className={styles.allergyTable}>
-          <thead>
-            <tr>
-              <td>ALLERGEN</td>
-              <td>
-                <div className={styles.centerItems}>
-                  SEVERITY & REACTION
-                  <svg className="omrs-icon" fill="rgba(0, 0, 0, 0.54)">
-                    <use xlinkHref="#omrs-icon-arrow-downward" />
-                  </svg>
-                </div>
-              </td>
-              <td>SINCE</td>
-              <td>UPDATED</td>
-            </tr>
-          </thead>
-          <tbody>
-            {patientAllergy &&
-              patientAllergy.entry.map(allergy => {
+  return (
+    <>
+      {patientAllergies?.length ? (
+        <SummaryCard
+          name="Allergies"
+          styles={{ width: "100%" }}
+          addComponent={AllergyForm}
+          showComponent={() =>
+            openWorkspaceTab(AllergyForm, "Allergies Form", {
+              allergyUuid: null
+            })
+          }
+        >
+          <table className={`omrs-type-body-regular ${styles.allergyTable}`}>
+            <thead>
+              <tr>
+                <td>ALLERGEN</td>
+                <td>
+                  <div className={styles.centerItems}>
+                    SEVERITY & REACTION
+                    <svg className="omrs-icon" fill="rgba(0, 0, 0, 0.54)">
+                      <use xlinkHref="#omrs-icon-arrow-downward" />
+                    </svg>
+                  </div>
+                </td>
+                <td>SINCE</td>
+                <td>UPDATED</td>
+              </tr>
+            </thead>
+            <tbody>
+              {patientAllergies.map(allergy => {
                 return (
-                  <React.Fragment key={allergy.resource.id}>
+                  <React.Fragment key={allergy?.id}>
```

Comment:
We could spare all the conditional members if we `filter` before we `map`.

Reply:
+1

Reply:
Though probably `patientAllergies` should never contain falsy members, and the conditional operator can just be removed.